run-make-check.sh

```diff
@@ -12,9 +12,27 @@
 #  version 2.1 of the License, or (at your option) any later version.
 #
 
-#
-# Return MAX(1, (number of processors / 2)) by default or NPROC
-#
+trap clean_up_after_myself EXIT
+trap clean_up_after_myself ERR
+
+ORIGINAL_CCACHE_CONF="$HOME/.ccache/ccache.conf"
+SAVED_CCACHE_CONF="$HOME/.run-make-check-saved-ccache-conf"
+SAVED_SOURCE_DATE_EPOCH="$SOURCE_DATE_EPOCH"
+
+function save_ccache_conf() {
+    test -f $ORIGINAL_CCACHE_CONF && cp $ORIGINAL_CCACHE_CONF $SAVED_CCACHE_CONF
+}
+
+function restore_ccache_conf() {
+    test -f $SAVED_CCACHE_CONF && mv $SAVED_CCACHE_CONF $ORIGINAL_CCACHE_CONF
+}
+
+function clean_up_after_myself() {
+    rm -fr ${CEPH_BUILD_VIRTUALENV:-/tmp}/*virtualenv*
+    export SOURCE_DATE_EPOCH="$SAVED_SOURCE_DATE_EPOCH"
```

Comment:
nit, i don't think user would want to `source` this script.

Reply:
Ah, now I get it. The `export` only exports to subshells. Thanks, fixing!